packages/typespec-python/test/generated/headasbooleantrue/headasbooleantrue/_operations/_operations.py

```diff
@@ -0,0 +1,813 @@
+        :return: bool
+        :rtype: bool
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+
+    @overload
+    def head_model(self, input: IO, *, content_type: str = "application/json", **kwargs: Any) -> bool:
+        """head_model.
+
+        :param input: Required.
+        :type input: IO
+        :keyword content_type: Body Parameter content-type. Content type parameter for binary body.
+         Default value is "application/json".
+        :paramtype content_type: str
+        :keyword bool stream: Whether to stream the response of this operation. Defaults to False. You
+         will have to context manage the returned stream.
+        :return: bool
+        :rtype: bool
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+
+    @distributed_trace
+    def head_model(self, input: Union[_models.VisibilityModel, JSON, IO], **kwargs: Any) -> bool:
+        """head_model.
+
+        :param input: Is one of the following types: VisibilityModel, JSON, IO Required.
+        :type input: ~headasbooleantrue.models.VisibilityModel or JSON or IO
+        :keyword content_type: Body parameter Content-Type. Known values are: application/json. Default
+         value is None.
+        :paramtype content_type: str
+        :keyword bool stream: Whether to stream the response of this operation. Defaults to False. You
+         will have to context manage the returned stream.
+        :return: bool
+        :rtype: bool
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+        error_map.update(kwargs.pop("error_map", {}) or {})
+
+        _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+        _params = kwargs.pop("params", {}) or {}
+
+        content_type: Optional[str] = kwargs.pop("content_type", _headers.pop("Content-Type", None))
+        cls: ClsType[None] = kwargs.pop("cls", None)
+
+        content_type = content_type or "application/json"
+        _content = None
+        if isinstance(input, (IOBase, bytes)):
+            _content = input
+        else:
+            _content = json.dumps(input, cls=AzureJSONEncoder)  # type: ignore
+
+        request = build_visibility_head_model_request(
+            content_type=content_type,
+            content=_content,
+            headers=_headers,
+            params=_params,
+        )
+        request.url = self._client.format_url(request.url)
+
+        _stream = kwargs.pop("stream", False)
+        pipeline_response: PipelineResponse = self._client._pipeline.run(  # pylint: disable=protected-access
+            request, stream=_stream, **kwargs
+        )
+
+        response = pipeline_response.http_response
+
+        if response.status_code not in [200]:
```

Comment:
That's incorrect generated code, if this is what we generate today from Swagger it's broken. For instance, a 404 should return False, and this code will make it an exception.

Reply:
@lmazuel In mgmt, only when [404 is defined](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/stable/2015-11-01/resources.json#L78-L114), we treat it as [normal status in generated code](https://github.com/Azure/azure-sdk-for-python/blob/3140a89fbcb8b521dc91d1af41565d572ef7b52b/sdk/resources/azure-mgmt-resource/azure/mgmt/resource/resources/v2016_02_01/aio/operations/_operations.py#L259-L265). So I think another option is: we keep current logic, and if users want 404 as False, let them define 404.